company.el

```diff
@@ -2909,10 +2916,10 @@ Delay is determined by `company-tooltip-idle-delay'."
 
 (defvar-local company-preview-overlay nil)
 
-(defun company-preview-show-at-point (pos)
+(defun company-preview-show-at-point (pos &optional company-common)
   (company-preview-hide)
 
-  (let ((completion (nth company-selection company-candidates)))
+  (let ((completion  (or company-common (nth company-selection company-candidates))))
```

Comment:
Sorry, I don't like this. This is pretty much the same definition as you showed initially. This function should take a new, non-optional argument.

And you can't thread it through `company-preview-frontend`: what if we want to add a new argument to all frontends later, that would do something else?

Instead, `company-preview-common-frontend` should not delegate to `company-preview-frontend` when COMMAND is `post-command`. It should call `company-preview-show-at-point` directly there.